### DISCUSSION TITLE
Thumbnail not available

### DISCUSSION THREAD

**Discussion by andreisimio** [structurizr/lite#146] [Q&A]
### Expected

I expect that the thumbnails are automatically refreshed

### Actual

after editing dls files, the diagrams thumbnails are not available

### Steps to reproduce

editing, diagrams does now update the thumbnails.

### Version/build information

I start using this command
#!/usr/bin/bash
sudo docker run -it --rm -p 8080:8080 -v ~/projects/arch-poc-structurizr:/usr/local/structurizr structurizr/lite

current build:
- build: 2024.06.18 (2024-06-18T10:11:32Z)

### Severity

Major

### Priority

I have no budget and there's no rush, please fix this for free

### More information

tried different approaches, to delete the hidden .structurizr folder in order to regenerate does not worked

**Comment by simonbrowndotje** [comment]
Thumbnails are only regenerated when you view/edit a diagram (https://docs.structurizr.com/ui/faq#why-arent-my-diagram-thumbnails-appearing) ... so I think this is working as expected.

**Comment by andreisimio** [comment]
[structurizr Thumbnails - replication .docx](https://github.com/user-attachments/files/15909736/structurizr.Thumbnails.-.replication.docx)

In the current instance, after editing a diagram, thumbnails are not regenerated . 
Attached is the replication document with screenshots and two approaches we tested (edit diagram and rename .structurizr hidden folder), still not found the solution on how to regenerate them.
Could you please suggest other solutions?

**Comment by simonbrowndotje** [comment]
> Steps to reproduce
> editing, diagrams does now update the thumbnails.

I think you may have a typo in the steps to reproduce ("now" -> "not") ... this is what caused me to say it seems to be working as expected.

In either case, perhaps take a look at the JavaScript/developer console in your web browser, particularly the XHR requests that save thumbnails to see if they are being blocked, or returning something other than HTTP 200.

Thumbnail generation works for me, so perhaps there's something in your workspace.dsl file that is causing an issue. Does thumbnail generation work with a simpler workspace?

**Comment by andreisimio** [comment]
Yes, for a simpler workspace, thumbnail generation works fine.
Now it became interesting, what could be the issues in the dsl file? Is there any option to review the model and have a debug on it?

**Comment by andreisimio** [comment]
SOLUTION: 
After commenting the branding/ logo lines in the  views / theme section, the thumbnails are automatically refreshed and works fine.

  /*    branding {
        logo "https://....-logo.jpg"    
    } 
*/

**Comment by simonbrowndotje** [comment]
I imagine you're running into an issue with CORS; see the following for more details:

- https://docs.structurizr.com/ui/diagrams/branding
- https://docs.structurizr.com/ui/diagrams/notation#icons
